Workaround gtk2hsC2hs death on glade-0.12.1 source:

gtk2hsC2hs: UName: root name supply used after saving

The "fix" is to reload binary instance after serialization.
Thah way we restore name counter.
diff -rN -u gtk2hs-buildtools-0.12.3/c2hs/toplevel/Main.hs gtk2hs-buildtools-0.12.3/c2hs/toplevel/Main.hs
--- gtk2hs-buildtools-0.12.3/c2hs/toplevel/Main.hs	2012-03-11 16:18:57.074032002 +0300
+++ gtk2hs-buildtools-0.12.3/c2hs/toplevel/Main.hs	2012-03-11 16:18:57.368032002 +0300
@@ -703,6 +703,8 @@
     -- save the attributed C to disk
     --
     liftIO $ putBinFileWithDict preCompFile (WithNameSupply cheader)
+    WithNameSupply cheader' <- liftIO $ getBinFileWithDict preCompFile
+    let _a = [cheader, cheader']
 
     --
     -- remove the pre-processed header
